teambuilder/web/views.py

```diff
@@ -1,90 +1,155 @@
+		email = request.POST.get('email')
+		password = request.POST.get('password')
+		password_repeat = request.POST.get('password_repeat')
+		mobile = request.POST.get('mobile')
+
+		if len(password) >= 8:
+			if password == password_repeat:
+				try:
+					with atomic():
+						user = User.objects.create_user(
+							username=student_id, email=email,
+							password=password)
+						user.first_name = firstname
+						user.last_name = lastname
+						user.save()
+						participant = models.Participant(user=user,
+														 mobile=mobile)
+						participant.save()
+						response = redirect('/')
+						response.set_cookie(key='username',
+											value=participant.user.username)
+						return response
+				except Exception as e:
+					error.append('Unable to create user - {0}'.format(e))
+			else:
+				error.append('Passwords do not match')
+		else:
+			error.append('Password length should be greater than 8')
+		if password != password_repeat:
+			error.append('Passwords do not match')
+	else:
+		error.append('Nothing provided')
+
+	return render(request, 'register.html', context={
+		'title': 'Register for Hackathon 2017',
+		'errors': error,
+		'user': None,
+	})
 
 
 def login(request):
-    # TODO: Add better checks
-    user = utils.get_login_user(request.COOKIES)
-    if user:
-        return redirect('/')
-    error = []
-    if request.POST.get('login', None):
-        username = request.POST.get('username')
-        password = request.POST.get('password')
-        login_users = User.objects.filter(username=username)
-        if len(login_users) > 0:
-            login_user = login_users[0]
-            if login_user.check_password(password):
-                response = redirect('/')
-                response.set_cookie(key='username', value=login_user.username)
-                return response
-        error.append('No such user found')
-    return render(request, 'login.html', context={
-        'title': 'Login',
-        'user': None,
-        'errors': error
-    })
+	# TODO: Add better checks
+	user = utils.get_login_user(request.COOKIES)
+	if user:
+		return redirect('/')
+	error = []
+	if request.POST.get('login', None):
+		username = request.POST.get('username')
+		password = request.POST.get('password')
+		login_users = User.objects.filter(username=username)
+		if len(login_users) > 0:
+			login_user = login_users[0]
+			if login_user.check_password(password):
+				response = redirect('/')
+				response.set_cookie(key='username', value=login_user.username)
+				return response
+		error.append('No such user found')
+	return render(request, 'login.html', context={
+		'title': 'Login',
+		'user': None,
+		'errors': error
+	})
+
+
+def register_team(request):
+	user = utils.get_login_user(request.COOKIES)
+	error = []
+	if user:
+		if request.POST.get('register', None):
+			team_name = request.POST.get('name')
+			# spaces are replaced by '-' due to security reasons
+			team_name = team_name.replace(' ', '-', 32)
```

Comment:
If you look at the regex that I've written, it doesn't allows spaces or any character that may be escaped as URL.